core/src/main/java/io/questdb/cairo/ColumnVersionWriter.java

```diff
@@ -275,6 +278,11 @@ private void doCommit() {
 
         Unsafe.getUnsafe().storeFence();
         storeNewVersion();
+
+        final int commitMode = configuration.getCommitMode();
+        if (commitMode != CommitMode.NOSYNC) {
+            mem.sync(commitMode == CommitMode.ASYNC);
```

Comment:
We should always sync metadata files if we want better stability on power loss.

Reply:
this is called from `commit()` and affects commit() latency. Same reasoning applies as here https://github.com/questdb/questdb/pull/3304#discussion_r1192785305

Reply:
How big is the latency increase? It shouldn't be very big considering that this method is called only when there are pending column version changes and, even when there are pending changes, the file itself is small.

Using msync/fsync for all metadata files and allowing the user to configure sync for data files seems to be a good tradeoff between power loss stability and an attempt to be fully durable. This will also make the logic easier to follow and document. WDYT?

Reply:
> But as you can see impact on frequent commit is very dramatic.

I wouldn't call 0.3ms dramatic considering that transactions typically include more than a single row. As for stability, as long as we extend column files in a durable way, the DB should recover on startup. Of course, the data might be corrupted which in the case of var-sized columns would mean potential crashes on data access.

Anyway, let's consider this comment to be a nit.